### PR TITLE
Modifications to enable Systems and Components to be exportable

### PR DESCRIPTION
**Pull request by bensanmorris**
Adds support for exportable components and systems (i.e. exporting components and systems from plugins). Have tested on Windows, macOS and Linux.

### CODE REVIEW

**Comment by maluramichael** [comment]
Maybe squash your commits. No one wants to review dozens of wip commits :D

**Comment by alecthomas** [comment]
Sorry, I looked at this a while back but forgot to submit my comments :\

Please revert all whitespace changes.

**Comment by bensanmorris** [comment]
No problem Alec - thanks for taking a look. I'll tidy up the whitespace and resubmit hopefully tmrw time permitting.

File: entityx/System.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "entityx/Entity.h"
 #include "entityx/Event.h"
 #include "entityx/help/NonCopyable.h"
+#include "Platform.h"
```
Comment (alecthomas) [inline entityx/System.h]: These includes should include the relative `entityx/Platform.h` path.

**Pull request closed without merging**